packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -406,14 +406,14 @@ class RenderParagraph extends RenderBox
     RenderBox child = firstChild;
     while (child != null) {
       final TextParentData textParentData = child.parentData;
-      final double scale = textParentData.scale;
-      Matrix4 localPosTrans = Matrix4.translation(Vector3(textParentData.offset.dx, textParentData.offset.dy, 0));
-      Matrix4 scaleTrans = Matrix4.diagonal3Values(scale, scale, scale);
-      Matrix4 transform = localPosTrans * scaleTrans;
+      final Matrix4 transform = Matrix4.translationValues(textParentData.offset.dx, textParentData.offset.dy, 0.0)
+        ..scale(textParentData.scale, textParentData.scale, textParentData.scale);
       final bool isHit = result.addWithPaintTransform(
         transform: transform,
         position: position,
         hitTest: (BoxHitTestResult result, Offset transformed) {
+          transform.invert();
+          assert(transformed == MatrixUtils.transformPoint(transform, position));
```

Comment:
What I meant was use your other method of computing the transformed point in the assert and see if it matches. (e.g. subtract the offset from the position and devide by scale or whatever is correct here and see if that matches transformed)